lib/containers/container_images.pm

```diff
@@ -78,14 +77,11 @@ sub build_and_run_image {
     # Test that we can execute programs in the container and test container's variables
     assert_script_run("$runtime run --rm --entrypoint 'printenv' myapp WORLD_VAR | grep Arda");
     assert_script_run("$runtime run -d --name myapp -p 8888:80 myapp");
-    script_retry("$runtime ps -a | grep myapp", delay => 5, retry => 3);    # ensure container is running
-    assert_script_run("$runtime logs myapp");    # show logs for easier problem investigation
+    validate_script_output_retry("$runtime ps -a", sub { m/myapp/ }, retry => 3, delay => 5, timeout => 300);
+    assert_script_run("$runtime logs myapp");
 
     # Test that the exported port is reachable
-    # FIPS libraries always check whether the host is in FIPS mode and must perform some initialization
-    # so it takes more time.
-    my $curl_opts = "--retry 500";
-    assert_script_run("curl $curl_opts http://localhost:8888/ | grep 'The test shall pass'");
+    validate_script_output_retry("curl -s http://localhost:8888/", sub { m/The test shall pass/ }, retry => 5, delay => 60, timeout => 300);
```

Comment:
No surprise that you need this. That's what I meant in your other PR that curl doesn't actually do the necessary retries :)